### PR TITLE
[Feature panel] Delete feature

### LINKED ISSUE #11: [Feature panel] Delete feature
Let's add the "Delete" entry to an overflow menu in the top-right of the feature details in the side panel when the feature is selected. Deleting the feature should also delete all related observations; we should show a BSW (big scary warning) confirmation dialog before doing this, i.e. "Deleting this map feature will delete all associated data, including any collected observations. This cannot be undone. Are you sure?" @sergeydolgov1 fyi/for comment.

### PR DESCRIPTION
**Pull request by gauravchalana1**
Closes #11.

Hi @gino-m 
Please review the Delete feature #11 function
Please share your comments on it and what can be improved.
A few pointers:
1. Where should be the positioning of this function?
2. I have commented on a few lines of prod code for testing of the function.

PFA screenshots
image
image

### CODE REVIEW

File: web-ng/src/app/services/data-store/data-store.service.ts
Comment (gino-m) [inline web-ng/src/app/services/data-store/data-store.service.ts:100]: Since .delete() itself returns a Promise, I think this `await` is redundant (it will unwrap the result from the Promise, which will get rewrapped by being returned from an `async` function.

File: web-ng/src/app/services/data-store/data-store.service.ts
Comment (gino-m) [inline web-ng/src/app/services/data-store/data-store.service.ts:117]: Same here.

File: web-ng/src/app/components/layer-list-item/layer-list-item.component.ts
Comment (gino-m) [inline web-ng/src/app/components/layer-list-item/layer-list-item.component.ts:37]: Any reason not to just define these two as optional with the `?` type modifier?

File: web-ng/src/app/components/layer-list-item/layer-list-item.component.ts
Comment (gino-m) [inline web-ng/src/app/components/layer-list-item/layer-list-item.component.ts:61]: We should add the subscriptions to a composite Subscription as we do in other classes and unsubscribe on ngOnDestroy().

**Review by gino-m — CHANGES_REQUESTED** [review]
Nice job, Gaurav, looks great so far!

Please place the delete option in an overflow menu (three dots) shown at the top-right of the feature panel (i.e., in the header next to the layer name) as described in #11. A few other comments inline. Thanks!

**Comment by gino-m** [comment]
@gauravchalana1 gentle ping! Note that tests are still failing, but I assume that's expected for the draft.

**Comment by gauravchalana1** [comment]
Hi @gino-m , yes I am supposing that the tests are failing because it’s a draft PR. Ty

**Comment by gauravchalana1** [comment]
Hi @gino-m , the changes have been made as you have suggested. PFB screenshots and share your reviews. ty :) 
image
image

File: web-ng/src/app/components/feature-panel/feature-panel.component.html
Comment (gino-m) [inline web-ng/src/app/components/feature-panel/feature-panel.component.html:36]: Looks like you reverted the change that adds the overflow menu (three dots). I assume this wasn't intentional?

File: web-ng/src/app/components/feature-panel/feature-panel.component.html
Comment (gino-m) [inline web-ng/src/app/components/feature-panel/feature-panel.component.html:51]: Out of scope of this PR, but while we're changing these lines anyway -  looks like we only use field_entry[1] , which contains the values in the `fields` map. Is there a way to just iterate over those instead? Maybe we can expose them as a property of the component?

File: web-ng/src/app/components/layer-list-item/layer-list-item.component.html
Comment (gino-m) [inline web-ng/src/app/components/layer-list-item/layer-list-item.component.html:28]: Let's put Delete as the last item in the menu.

File: web-ng/src/app/components/layer-list-item/layer-list-item.component.ts
Comment (gino-m) [inline web-ng/src/app/components/layer-list-item/layer-list-item.component.ts:38]: Do we still need `| null` with `?`?

**Review by gino-m — CHANGES_REQUESTED** [review]
Nice work! A few suggestions, otherwise LG!

File: web-ng/src/app/components/feature-panel/feature-panel.component.html
Comment (gauravchalana1) [reply web-ng/src/app/components/feature-panel/feature-panel.component.html:36]: It was intentional for testing this feature, fixed it. ty :)

File: web-ng/src/app/components/layer-list-item/layer-list-item.component.ts
Comment (gauravchalana1) [reply web-ng/src/app/components/layer-list-item/layer-list-item.component.ts:38]: yes we do need | null with ? as in the router service we are projectId?: string | null because paramap returns an observable of string | null type.

File: web-ng/src/app/components/feature-panel/feature-panel.component.html
Comment (gino-m) [reply web-ng/src/app/components/feature-panel/feature-panel.component.html:51]: I tried to fix this directly in your PR, but it looks like I don't have write access to your branch; I'll send a follow-up PR once this is merged.

**Review by gino-m — APPROVED** [review]
LGTM.. 🚀  Nice work! 💯

**Pull request merged**
